server/src/service.rs

```diff
@@ -410,4 +433,23 @@ impl Service {
 
         n_sessions
     }
+
+    pub async fn prompt_index(&self) -> u32 {
+        let n_prompts = *self.prompt_index.read().await + 1;
+        *self.prompt_index.write().await = n_prompts;
+
+        n_prompts
+    }
+
+    pub async fn prompt(&self) -> Option<Prompt> {
+        let prompts = self.prompts.lock().await;
+
+        prompts.last().cloned()
+    }
+
+    pub async fn remove_prompt(&self) {
+        // prompts should always contain one item during a prompt related operation and
+        // it should be cleaned up afterwards.
+        self.prompts.lock().await.pop();
+    }
```

Comment:
If there would always be one single prompt (which i don't believe is true, except for maybe GNOME specifically) as the API clearly allows modal keyring dialogs so every app could have an ongoing prompt operation.

Reply:
Yeah, this is correct and I have removed this restriction.